core/repository/api/src/main/java/org/eclipse/rdf4j/repository/Repository.java

```diff
@@ -99,12 +100,21 @@ default void init() throws RepositoryException {
 	 * @return A connection that allows operations on this repository.
 	 * @throws RepositoryException If something went wrong during the creation of the Connection.
 	 */
-	public RepositoryConnection getConnection() throws RepositoryException;
+	RepositoryConnection getConnection() throws RepositoryException;
 
 	/**
 	 * Gets a ValueFactory for this Repository.
 	 *
 	 * @return A repository-specific ValueFactory.
 	 */
-	public ValueFactory getValueFactory();
+	ValueFactory getValueFactory();
+
+	default RepositorySettings getSettings() {
+		throw new UnsupportedOperationException();
+	}
+
+	default void setSettings(RepositorySettings settings) {
+		throw new UnsupportedOperationException();
+	}
```

Comment:
I don't understand why you're adding this to the Repository interface. Didn't we want a configuration option for a Repository_Connection_ ? We already have an existing configuration model for repositories and sail stacks - the use case was inspired by needing to tweak options on the fly, per connection (or even per request).